src/components/Tables/Table.tsx

```diff
@@ -13,22 +13,21 @@ import {
 } from "@fortawesome/free-solid-svg-icons";
 import IconButton from "@mui/material/IconButton/IconButton";
 import Icon from "@/components/Icons/Icon";
+import { Filter, headerLabelFromKey, textFilter } from "@/components/Tables/Filters";
 
-export interface Datum {
-    [headerKey: string]: string | number | boolean | null;
-}
-
-export type TableHeaders = [string, string][];
+export type TableHeaders<Data> = readonly (readonly [keyof Data, string])[];
```

Comment:
Why readonly (readonly ?

Reply:
I need to be able to accept deeply immutable data here to accept `as const`